### PR TITLE
Add option group for setting bank/background bins and reading loudest triggers from bins from CLI

### PR DESCRIPTION
**Pull request by cmbiwer**

### CODE REVIEW

**Comment by cmbiwer** [comment]
Ended up adding another CLI function for getting the loudest triggers from coinc or sngl trigger files.

Example that I tested with on sugwg:
```
echo "
#! /usr/bin/env python

import argparse
from pycbc.io import hdf

parser = argparse.ArgumentParser()
hdf.insert_loudest_triggers_option_group(parser)
opts = parser.parse_args()

names, data = hdf.loudest_triggers_from_cli(
                                    opts, coinc_parameters=['stat'],
                                    sngl_parameters=['snr', 'end_time'],
                                    bank_parameters=['mass1', 'mass2'])
print names
print data
" > test_cli.py

RUN_DIR="/home/amber.lenon/projects/cbc/o1_Analysis/Analysis-rerun/analysis3-C01-no_injections-rerun/output"
BANK_FILE="${RUN_DIR}/bank/H1L1-BANK2HDF-1128299417-1083600.hdf"
STATMAP_FILE="${RUN_DIR}/full_data/H1L1-COMBINE_STATMAP_FULL_DATA_FULL_CUMULATIVE_CAT_12H_FULL_DATA_FULL-1128299417-1083600.hdf"
SNGL_TRIGGER_FILES="H1:${RUN_DIR}/full_data/H1-HDF_TRIGGER_MERGE_FULL_DATA-1128299417-1083600.hdf L1:${RUN_DIR}/full_data/L1-HDF_TRIGGER_MERGE_FULL_DATA-1128299417-1083600.hdf"

# compare to https://sugwg-jobs.phy.syr.edu/~amber.lenon/o1/analysis3-C01-no_injections/analysis3-c01-1.3.6-no_injections/7._open_box_result/
echo "Do coinc test"
python test_cli.py \
    --bank-file ${BANK_FILE} \
    --statmap-file ${STATMAP_FILE} \
    --sngl-trigger-files ${SNGL_TRIGGER_FILES} \
    --search-n-loudest 30 \
    --n-loudest 3 \
    --bank-bins b1:total:lt4 b2:total:lt150

# compare to https://sugwg-jobs.phy.syr.edu/~amber.lenon/o1/analysis3-C01-no_injections/analysis3-c01-1.3.6-no_injections/3._single_triggers/3.05_H1_loudest_triggers/
echo "Do sngl test"
SNGL_TRIGGER_FILES="H1:${RUN_DIR}/full_data/H1-HDF_TRIGGER_MERGE_FULL_DATA-1128299417-1083600.hdf"
VETO_FILE="${RUN_DIR}/segments/H1L1-FOREGROUND_CENSOR-1128299417-1083600.xml"
VETO_SEGMENT_NAME="closed_box"
python test_cli.py \
    --bank-file ${BANK_FILE} \
    --sngl-trigger-files ${SNGL_TRIGGER_FILES} \
    --bank-bins all:total:lt150 \
    --search-n-loudest 30 \
    --n-loudest 3 \
    --veto-file ${VETO_FILE} \
    --veto-segment-name ${VETO_SEGMENT_NAME}
```

So now with a couple lines can get a dict of coinc/sngl/bank data of loudest events in background bins using CLI.

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -16,7 +16,9 @@
 from pycbc_glue.ligolw import utils as ligolw_utils
 from pycbc_glue.ligolw.utils import process as ligolw_process
 
+from pycbc import types
```
Comment (ahnitz) [inline pycbc/io/hdf.py]: I think this might belong better in pycbc/events somwhere. What do you think?

Comment (cmbiwer) [inline pycbc/io/hdf.py]: Perhaps its most closely related to ``coinc`` but the function is a bit broader than coinc triggers. Could add a ``pycbc/events/triggers.py`` and put it in there and move the background/bank bin CLI function to there as well. Unless you have another suggestion.

Comment (ahnitz) [inline pycbc/io/hdf.py]: That seems like a reasonable solution to me.

**Comment by cmbiwer** [comment]
I've moved the functions into a new module ``pycbc.events.triggers``.

**Comment by spxiwh** [comment]
@cmbiwer Please check this to see if these are genuine issues:

https://landscape.io/github/ligo-cbc/pycbc/1699/diff

**Pull request merged**